### ISSUE TITLE
Popp IP44 Keypad is not sending all codes reliably

### ISSUE THREAD

**Issue by Mastiffen** [zwave-js/zwave-js#2887]
**Checklist:**

-   [x ] My problem is **not** within Home Assistant or the ZWave JS integration. **Or:** a Home Assistant developer has told me to come here.
-   [ x] My problem is **not** within ZWaveJS2MQTT. **Or:** a ZWaveJS2MQTT developer has told me to come here.
-   [ x] I have checked the troubleshooting section and my problem is **not** described there.
-   [ x] I have read the changelog and my problem was **not** mentioned there.

**Describe the bug**

A clear and concise description of what the bug is. Describe what causes the bug, what you observe, and what happens as a result.

All codes are not received. It seems like they are blocked arbitrary. The only code that works all the time is user code 3.

**Device information**

Which device(s) is/are affected (manufacturer/model)?  
What is/are the node IDs?
Popp IP44 keyboard, node ID 11

**Did you change anything?**

-   [x ] **Yes**: _(please describe)_
-   [ ] **No**
I did re-interview. First time the whole user code section dissappeared. Second time it came back.

**Did this use to work before?**

-   [ x] **Don't know**, this is a new device
-   [ ] **No**, it never worked anywhere
-   [ ] **Yes**, in: _(specify application with versions)_

**How are you using `node-zwave-js`**

-   [ ] `zwavejs2mqtt` (latest) docker image
-   [ ] `zwavejs2mqtt` (dev) docker image
-   [ ] `zwavejs2mqtt` Manual Docker build
    -   `node-zwave-js` branch: 
    -   `zwavejs2mqtt` branch: 
-   [ ] ioBroker.zwave2 adapter
-   [ ] `HomeAssistant` version XYZ
-   [ ] Pkg
-   [ ] Manual Docker build
    -   `node-zwave-js` branch: 
    -   `zwavejs2mqtt` branch: 
-   [ ] `node-red-contrib-zwave-js`
    -   Module version (double click node): 
-   [x ] Manually built (as described in the [docs](https://zwave-js.github.io/node-zwave-js/#/development/installing-from-github))
-   [ ] Other: 

**To Reproduce**

2. Click on '...'
3. Scroll down to '...'

**Additional context**

**Logfile:**

There is no logfile to send, there isn't anything in the log when this happens.

**Comment by AlCalzone** [comment]
> there isn't anything in the log when this happens

Then the messages from the keypad are not reaching the Z-Wave controller. Unreliable communication is often the result of a bad controller placement and/or interference. Try putting it on an USB extension cord, away from metallic surfaces. USB ports (on a RPi for example) are sometimes enough to interfere.

**Comment by Mastiffen** [comment]
Thanks, but the keypad is around five meters from the USB controller, with two 15 cm wooden walls and no metal in between, no water pipes and no power cables. The Z-Wave USB device is on an unpowered USB hub around 50 cm from the Pi and anything else that can interfere, and I have a bunch of other stuff around the cabin that has repeater capabilities. There are two Z-Wave dimmers in the light fixtures in the room where the Popp is on the outer wall, less than two meters away. It may of course be that the Popp has a ridiculously bad range. I have put a repeating plug-in switch on the inside of the outer wall where the Popp is on the inside and I will see if that helps. But I doubt it, since the dimmers should be repeating too.

Oh, and I have another Popp at home, and that has been giving me problems with OZW, but I haven't tested it on the JS version yet.

**Comment by AlCalzone** [comment]
Distance is not always a factor. I'm not an expert on the radio side of Z-Wave, but from my experience, even moving the controller a few cm can sometimes improve the situation. That being said, maybe it's the Popp that's in a bad spot.

One way to figure that out is by using a Zniffer next to the controller and next to the Popp and to capture the RF traffic that's visible in each location - although that's also no gurarantee the other devices see the same.

If you haven't already, you can also try healing the network.

**Comment by Mastiffen** [comment]
I will try to do a new healing. But still with repeaters all around the place (almost all non-battery powered Z-wave devices have the repeater capability, but you most certainly knew that)  and even about two meters from the Popp, with only 15 cm of wood between them the place ment of the actual controller doesn't matter too much. That's the very nice thing about Z-Wave, I have a house with a lot of places were I can't get 433mHz, but Z-Wave works without a hitch. The Namron dimmers are marked as repeaters, but it may be that the repeating capabilities are weak.

**Comment by AlCalzone** [comment]
All I can do here is try and give you some hints. If the controller doesn't see the messages, it's definitely not a software issue.

**Comment by Mastiffen** [comment]
OK, thanks for the help! So even if it sees one reliably (Scene 003) and the other ones not reliably that isn't the software's fault. If you are sure that implementation works I will do some hardware testing with placements of repeaters. :)

**Comment by AlCalzone** [comment]
> So even if it sees one reliably (Scene 003) and the other ones not reliably

To be sure: which log are you looking at? The driver log should at least include the raw serial data if something is received.

**Comment by Mastiffen** [comment]
I was looking at the debug info in the node. From your question I suspect that's not the log I am looking for.

Edit: And a heal made most of my regular, powered nodes fail, even if they seem to have updated their neighbours. Only a few battery powered nodes were green. Something's not right here, I think.

**Comment by Mastiffen** [comment]
Of these only the "Garasjevinsj" should fail because it's not plugged in right now: 

![image](https://user-images.githubusercontent.com/35197020/122650538-dca0e000-d133-11eb-860b-92d6a5606536.png)

**Comment by AlCalzone** [comment]
> I suspect that's not the log I am looking for.

I'm looking for the one that's requested in the issue template:
https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/bug_report?id=driver-logs

**Comment by Mastiffen** [comment]
Aha, I misunderstood something, because I never got to that page. Btw I see that something's wrong here, because even after two heals (which fixed most of the stuff to green) the Popp doesn't seem to have any neighbours. But maybe it's not supposed to?

![image](https://user-images.githubusercontent.com/35197020/122650756-eaa33080-d134-11eb-94da-439924d1b895.png)

I apologize again for misunderstanding, but I have activated the log now and I will be back tomorrow with log, when I have the time to test some more.

**Comment by Mastiffen** [comment]
I have now tested sporadically in three days because I'm totally confused. The healing I did on the Sunday didn't change anything. But the next morning, when I tried to test, all codes worked! And nothing physically has changed, no nodes are moved or anything. Also I have found out that this device isn't supposed to keep track of neighbours because it's considered a "moveable" device, which makes no sense since it's heavy and meant to be bolted to the wall. My Namron 4 channel wall switches do keep track of neigbours, even if they are small and light.

Anyway, I will keep this closed and then get back to it if the messy behaviour starts again.
